x-pack/plugin/mapper-unsigned-long/src/main/java/module-info.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+module org.elasticsearch.mapper.unsigned_long {
+    requires org.apache.lucene.core;
+    requires org.elasticsearch.server;
+    requires org.apache.lucene.sandbox;
+    requires org.elasticsearch.xcontent;
+    requires org.elasticsearch.painless.spi;
+
+    exports org.elasticsearch.xpack.unsignedlong; // for the painless script engine
```

Comment:
I remember!  The `mapper.unsignedlong` and `painless` modules are siblings, whose common parent is the `painless.spi` module. To support qualified exports across these to modules, then the plugin loader must orchestrate the qualified export - it will not "just" happen when defining the `mapper.unsignedlong`, since the target of the qualified export is not in an ancestor module layer. I'll do this as a followup, if that is ok.

Reply:
Yeah, I don't think it's worth jumping through hoops for this.